lib/libzfs/libzfs_pool.c

```diff
@@ -3852,8 +3853,13 @@ zpool_history_unpack(char *buf, uint64_t bytes_read, uint64_t *leftover,
 		/* add record to nvlist array */
 		(*numrecords)++;
 		if (ISP2(*numrecords + 1)) {
-			*records = realloc(*records,
+			tmp = realloc(*records,
```

Comment:
In the case of an error records must be considered undefined so you don't strictly need the tmp variable here.  The original line can remain safely.